assembly/src/assembler/instruction/crypto_ops.rs

```diff
@@ -4,6 +4,48 @@ use vm_core::{AdviceInjector, Decorator};
 // HASHING
 // ================================================================================================
 
+/// Appends HPERM and stack manipulation operations to the span block as required to compute a
+/// 1-to-1 hash. The top of the stack is expected to be arranged with 1 word (4 elements) to be
+/// hashed: [A, ...]. The resulting stack will contain the 1-to-1 hash result: [C, ...].
+///
+/// This assembly instruction uses the VM operation HPERM at its core, which permutes the top 12
+/// elements of the stack.
+///
+/// To perform the operation we do the following:
+/// 1. Prepare the stack with 12 elements for HPERM by pushing 4 more elements for the capacity,
+///    then reordering the stack and pushing an additional 4 elements so that the stack looks
+///    like: [0, 0, 0, 0, a3, a2, a1, a0, 0, 0, 0, 0, ...].  All capacity elements and the highest
+///    most rate elements are set to ZERO, in accordance with the RPO rules.
+/// 2. Append the HPERM operation, which performs a permutation of RPO on the top 12 elements and
+///    leaves the an output of [D, C, B, ...] on the stack.  C is our 1-to-1 has result.
+/// 3. Drop D and B to achieve our result [C, ...]
+pub(super) fn hash(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
+    #[rustfmt::skip]
+    let ops = [
+        // add 4 elements to the stack to be used as the capacity elements for the RPO permutation
+        Pad, Pad, Pad, Pad,
```

Comment:
There is one issue here: since we are hashing 4 elements (which is not a multiple of 8), according to RPO specs we should set the first capacity element to `ONE` rather than `ZERO`.

Reply:
updated.  I also had to push `ONE` as the first padded element in the rate.  Tests are passing now.